pkg/blob/azure_msi.go

```diff
@@ -0,0 +1,96 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blob
+
+// This code has been copy-pasted from https://github.com/google/go-cloud/blob/master/blob/azureblob/azureblob.go
+// because the gocloud.dev package is pinned to version v0.19.0, which doesn't support MSI.
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"gocloud.dev/blob/azureblob"
+)
+
+const (
+	tokenRefreshTolerance = 300
+)
+
+// openerFromMSI acquires an MSI token and returns TokenCredential backed URLOpener
+func openerFromMSI(accountName azureblob.AccountName, clientId ClientID, opts azureblob.Options) (*azureblob.URLOpener, error) {
+
+	spToken, err := getMSIServicePrincipalToken(azure.PublicCloud.ResourceIdentifiers.Storage, clientId)
+	if err != nil {
+		return nil, fmt.Errorf("failure acquiring token from MSI endpoint %w", err)
```

Comment:
If this file is here to stay, use TTS errors

Reply:
I'll change these lines to just `return nil, err`.